specification/resources/resource-manager/Microsoft.Authorization/stable/2023-04-01/policyDefinitionVersions.json

```diff
@@ -0,0 +1,899 @@
+          },
+          {
+            "name": "policyDefinitionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[^<>*%&:\\?.+/]*[^<>*%&:\\?.+/ ]+$",
+            "description": "The name of the policy definition."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/TopParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns an array of policy definition versions.",
+            "schema": {
+              "$ref": "#/definitions/PolicyDefinitionVersionListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CloudError": {
+      "type": "object",
+      "x-ms-external": true,
+      "properties": {
+        "error": {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ErrorResponse"
+        }
+      },
+      "description": "An error response from a policy operation."
+    },
+    "PolicyDefinitionVersionProperties": {
+      "type": "object",
+      "properties": {
+        "policyType": {
+          "type": "string",
+          "description": "The type of policy definition. Possible values are NotSpecified, BuiltIn, Custom, and Static.",
+          "enum": [
+            "NotSpecified",
+            "BuiltIn",
+            "Custom",
+            "Static"
+          ],
+          "x-ms-enum": {
+            "name": "policyType",
+            "modelAsString": true
+          }
+        },
+        "mode": {
+          "type": "string",
+          "description": "The policy definition mode. Some examples are All, Indexed, Microsoft.KeyVault.Data.",
+          "default": "Indexed"
+        },
+        "displayName": {
+          "type": "string",
+          "description": "The display name of the policy definition."
+        },
+        "description": {
+          "type": "string",
+          "description": "The policy definition description."
+        },
+        "policyRule": {
+          "type": "object",
+          "description": "The policy rule."
+        },
+        "metadata": {
+          "type": "object",
+          "description": "The policy definition metadata.  Metadata is an open ended object and is typically a collection of key value pairs."
+        },
+        "parameters": {
+          "description": "The parameter definitions for parameters used in the policy rule. The keys are the parameter names.",
+          "$ref": "#/definitions/ParameterDefinitions"
+        },
+        "version": {
```

Comment:
Could all of this be replaced by a $ref to ./policyDefinitions.json#/definitions/PolicyDefinitionProperties? Maybe there are some differences, but it looks like it would benefit from factoring out and sharing the common properties. Not essential/blocking, just a hygiene suggestion.

Reply:
they are not exactly the same payload so it can't be referenced directly.  This a slight subset of that.